### PR TITLE
Add infer-conventions skill to record app conventions

### PR DESCRIPTION
**Pull request by pushpak1300**
Currently a new Boost user starts with an empty `.ai/rules/` and has to hand write the conventions their codebase already follows. Boost can record rules one at a time with the `record-rule` MCP tool, but nothing bootstraps them from how the app is actually written.

This PR ships an `infer-conventions` skill. Instead of a deterministic command, it drives the agent to sweep the app across ~49 Laravel convention dimensions (validation, models, architecture, testing, frontend, database), discover the house patterns a fixed detector can't name (Action classes, query objects, module layout, deliberate non-patterns), and record the real decisions as path-scoped rules via `record-rule`.

Run the skill and it maps your `app/` tree, works each detection, shows the candidates with evidence, and records the ones you approve. `boost:install` now nudges you to run it at the end.

### CODE REVIEW

File: src/Console/InstallCommand.php [hunk 1]
```diff
@@ -92,6 +93,11 @@ public function handle(): int
         $this->discoverEnvironment();
         $this->collectInstallationPreferences();
         $this->performInstallation();
+
+        if ($this->selectedBoostFeatures->contains('skills')) {
```
Comment (WendellAdriel) [inline src/Console/InstallCommand.php]: This fires whenever the `skills` feature is selected, so it still shows when `infer-conventions` was excluded via `boost.skills.exclude` or when no skills-capable agent received the skill, pointing the user at a skill they don't have. Let's gate it on `in_array('infer-conventions', $this->installedSkillNames, true)` instead (it already reflects exclusions and defaults to `[]`), and add an install-command test asserting the note shows when the skill is installed and stays silent when it isn't.

File: tests/Unit/Install/SkillWriterTest.php [hunk 2]
```diff
@@ -36,6 +55,82 @@ function cleanupSkillDirectory(string $path): void
...
+    $relativeTarget = '.boost-test-skills-'.uniqid();
+    $absoluteTarget = base_path($relativeTarget);
+
+    writeInferConventions($relativeTarget);
+
+    $skill = file_get_contents($absoluteTarget.'/infer-conventions/SKILL.md');
+    $checklist = file_get_contents($absoluteTarget.'/infer-conventions/references/checklist.md');
+
+    expect($skill)
+        ->not->toContain('app/Livewire/**')
```
Comment (WendellAdriel) [inline tests/Unit/Install/SkillWriterTest.php]: `app/Livewire/**` never appears in either template in any render branch, so this assertion can never fail and implies the Livewire-installed render includes that glob when it doesn't. Let's drop it.

File: tests/Unit/Install/SkillWriterTest.php [hunk 3]
```diff
@@ -3,8 +3,27 @@
...
 
 use Laravel\Boost\Contracts\SupportsSkills;
+use Laravel\Boost\Install\GuidelineAssist;
+use Laravel\Boost\Install\GuidelineConfig;
 use Laravel\Boost\Install\Skill;
 use Laravel\Boost\Install\SkillWriter;
+use Laravel\Roster\PackageCollection;
+use Laravel\Roster\ProjectManager;
+
+function writeInferConventions(string $relativeTarget): void
```
Comment (WendellAdriel) [inline tests/Unit/Install/SkillWriterTest.php]: This helper duplicates the body of the first test below and throws away the `SkillWriter::write()` result, so the two render tests never confirm the write succeeded before reading the output files. A failure would surface as a confusing "failed to open stream" instead of a clear assertion. Let's have it return the `int` result, reuse it in the first test, and assert `SkillWriter::SUCCESS` in both render tests.

File: .ai/boost/skill/infer-conventions/references/checklist.blade.php [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+   - Hint: typed model params in signatures vs `findOrFail(` in controllers; grep `Route::bind`.
+10. Rate limiting: named `RateLimiter::for()` + `throttle:name` vs inline `throttle:60,1`.
+    - Hint: grep `RateLimiter::for` in providers vs `throttle:` in route files.
+
+## C. Authorization
+
+11. Authorization home: Gates (`Gate::define`) vs Policy classes in `app/Policies`.
+    - Hint: `ls app/Policies`; grep `Gate::define` in `app/Providers`.
+12. Authorization call site: `$this->authorize()` / `Gate::authorize()` vs `$user->can()` vs `can` middleware vs `#[Authorize]` vs `@can` in Blade.
+    - Hint: grep `authorize(`, `->can(`, `middleware('can:`, `#[Authorize(`, `@can(`.
```
Comment (WendellAdriel) [inline .ai/boost/skill/infer-conventions/references/checklist.blade.php:43]: The literal `@can(`, `@include`, and `@props` tokens only survive Blade rendering because the backtick placeholder in `RendersBladeGuidelines` glues a word character before the `@`, making Blade's `\B@` directive regex skip them. A future edit that puts one of these tokens after a space inside a code span would get compiled and corrupt the rendered checklist, and no test would notice. Let's add a positive assertion that the rendered checklist still contains these literal tokens, or add them to the placeholder map like `@volt`.

File: src/Install/SkillComposer.php [hunk 5]
```diff
@@ -53,12 +53,26 @@ public function skills(): Collection
         $excluded = config('boost.skills.exclude', []);
 
         return $this->skills = collect()
+            ->merge($this->getCoreSkills())
```
Comment (WendellAdriel) [inline src/Install/SkillComposer.php:56]: With core skills now shipping to every project unconditionally, `boost.skills.exclude` becomes the only opt-out, and it isn't in the published config, README, or docs. Let's add a `skills.exclude` entry to `config/boost.php` or document it in the README so users can discover it.

File: tests/Feature/Console/ListSkillCommandTest.php [hunk 6]
```diff
@@ -6,6 +6,7 @@
 
 beforeEach(function (): void {
     File::deleteDirectory(base_path('.ai/skills'));
+    config(['boost.skills.exclude' => ['infer-conventions']]);
```
Comment (WendellAdriel) [inline tests/Feature/Console/ListSkillCommandTest.php:10]: It isn't obvious why every test in this file excludes the new skill. Let's add a one-line comment saying this keeps the tests exercising the empty and user-skill states now that a core skill always ships.

**Review by WendellAdriel — CHANGES_REQUESTED** [review]
The PR body says the nudge happens "at the end of an interactive install", but it also shows on non-interactive runs since the check never looks at interactivity. Let's either update the description or add the interactivity check if interactive-only was the intent.

File: .ai/boost/skill/infer-conventions/references/checklist.blade.php (hunk 4 above)
Comment (pushpak1300) [reply .ai/boost/skill/infer-conventions/references/checklist.blade.php:43]: make sense.

File: src/Install/SkillComposer.php (hunk 5 above)
Comment (pushpak1300) [reply src/Install/SkillComposer.php:56]: i think this is intentional. will plan adding it seprately.

**Review by WendellAdriel — APPROVED** [review]
LGTM :muscle:
